Add information about different types of ACS (#1971)
theforeman/foreman-documentation: guides/common/modules/con_managing-alternate-content-sources.adoc

= Managing Alternate Content Sources

Alternate content sources define alternate paths to download content during synchronization.

> The content itself is downloaded from the alternate content source and only the metadata is downloaded from the {ProjectServer}.

Reviewer: Metadata can come from the "upstream URL" or Foreman depending on the situation.  If the ACS is created on the Foreman's main smart proxy, then the metadata will come from some upstream server.  If the ACS is created on a remote smart proxy, then the metadata will come from the Foreman.

I suppose you could just say something like "upstream server or Foreman server depending on the configuration".

You can use alternate content source to speed up synchronization if the content is located on the local filesystem or on a nearby network.
You can set up alternate content sources for {ProjectServer} and {SmartProxy}.
You must refresh the alternate content source after creation or after making any changes.
A weekly cron job refreshes all alternate content sources.
You can also refresh the alternate content sources manually using the {ProjectWebUI} or the Hammer CLI.

There are three types of alternate content sources:

Custom::
Custom alternate content source downloads the content from any upstream repository or from the local filesystem.

Simplified::
Simplified alternate content source copies the upstream repo information from the {ProjectServer}.

> You can use simplified alternate content sources in situations where connections to the upstream repo is faster than the {ProjectServer}.

Reviewer: ```suggestion
You can use simplified alternate content sources in situations where the connection from the {SmartProxy} to the upstream repo is faster than the {ProjectServer}.
```

= Managing Alternate Content Sources

Alternate content sources define alternate paths to download content during synchronization.
The content itself is downloaded from the alternate content source, while the metadata is downloaded from the {ProjectServer} or the upstream URL, depending on the configuration.
You can use alternate content source to speed up synchronization if the content is located on the local filesystem or on a nearby network.
You can set up alternate content sources for {ProjectServer} and {SmartProxy}.
You must refresh the alternate content source after creation or after making any changes.
A weekly cron job refreshes all alternate content sources.
You can also refresh the alternate content sources manually using the {ProjectWebUI} or the Hammer CLI.

There are three types of alternate content sources:

Custom::

> Custom alternate content source downloads the content from any upstream repository or from the local filesystem.

Reviewer 2: ```suggestion
Custom alternate content source downloads the content from any upstream repository or the local filesystem.
```
Avoiding repetition.

Simplified::

> Simplified alternate content source copies the upstream repo information from the {ProjectServer}.

Reviewer 3: ```suggestion
Simplified alternate content source copies the upstream repo information from your {ProjectServer}.
```

> You can use simplified alternate content sources in situations where the connection from the {SmartProxy} to the upstream repo is faster than the {ProjectServer}.

Reviewer 3: ```suggestion
You can use simplified alternate content sources in situations where the connection from your {SmartProxy} to the upstream repo is faster than from your {ProjectServer}.
```

ifdef::satellite[]

> Selecting the Red Hat products when creating a simplified alternate content source will download the content to the {SmartProxies} from the Red Hat CDN.

Reviewer 3: ```suggestion
Selecting the Red Hat products when creating a simplified alternate content source will download the content to your {SmartProxies} from the {Team} CDN.
```

"Team" attribute contains a non breaking whitespace.

endif::[]

RHUI::

> RHUI alternate content source downloads content from the Red Hat Update Infrastructure server in your environment.

Reviewer 3: ```suggestion
RHUI alternate content source downloads content from the Red Hat Update Infrastructure server in your environment.
```

Is it "in" your infrastructure or "to" your infrastructure?

Author: Hey @maximiliankolb it has to be "in" your infrastructure as we are talking about the RHUI setup in the user's environment. Should I rephrase it?

ifdef::satellite[]

> Note that Red{nbsp}Hat only supports RHUI type alternate content source with RHUI 4 or above.

Reviewer 3: ```suggestion
{Team} only supports RHUI type alternate content source with RHUI 4 or above.
```

> {Team} also does not support configuration of RHUI alternate content source with the {Team} provided RHUI instances for AWS.

Reviewer 2: ```suggestion
{Team} also does not support the configuration of RHUI alternate content source with the {Team} provided RHUI instances for AWS.
```

> {Team} also does not support configuration of RHUI alternate content source with the {Team} provided RHUI instances for AWS.

Reviewer: Rather than saying that AWS RHUI is not supported, it might be better to say that only the default RHUI installation is supported. There might in the future be other modes of "special" RHUI installations that Satellite also won't support. Here's a comment with the original explanation: https://bugzilla.redhat.com/show_bug.cgi?id=2001087#c6

> Red{nbsp}Hat also does not support configuration of RHUI alternate content source with the Red{nbsp}Hat provided RHUI instances for AWS.

Reviewer 3: ```suggestion
{Team} also does not support configuration of RHUI alternate content source with the {Team} provided RHUI instances for AWS.
```

= Managing Alternate Content Sources

Alternate content sources define alternate paths to download content during synchronization.

> The content itself is downloaded from the alternate content source, while the metadata is downloaded from the {ProjectServer} or the upstream URL, depending on the configuration.

Reviewer: ```suggestion
The content itself is downloaded from the alternate content source, while the metadata is downloaded from the {ProjectServer} or the upstream URL, depending on the configuration.
Alternate content source paths consist of a base URL and any number of appended subpaths. If your base URL is https://server.example.com and your subpaths are rhel7/ and rhel8/, then both https://server.example.com/rhel7/ and https://server.example.com/rhel8/ will be searched.
```

Author: > Alternate content source paths consist of a base URL and any number of appended subpaths. If your base URL is https://server.example.com and your subpaths are rhel7/ and rhel8/, then both https://server.example.com/rhel7/ and https://server.example.com/rhel8/ will be searched.

@ianballou does this apply to both RHUI and custom ACS?

Reviewer: It does indeed.

Author: Hey @ianballou I added this information to the RHUI and Custom ACS prerequisites.

You can use alternate content source to speed up synchronization if the content is located on the local filesystem or on a nearby network.
You can set up alternate content sources for {ProjectServer} and {SmartProxy}.
You must refresh the alternate content source after creation or after making any changes.
A weekly cron job refreshes all alternate content sources.
You can also refresh the alternate content sources manually using the {ProjectWebUI} or the Hammer CLI.

There are three types of alternate content sources:

Custom::

> Custom alternate content source downloads the content from any upstream repository or the local filesystem.

Reviewer: I'd like to make it clear that the filesystem path must also be a repository:

```suggestion
Custom alternate content sources download content from any upstream repository on the network or filesystem.
```

Simplified::

> Simplified alternate content source copies the upstream repo information from your {ProjectServer}.

Reviewer: ```suggestion
Simplified alternate content sources copy the upstream repository information from your {ProjectServer} for the selected products.
```

> You can use simplified alternate content sources in situations where the connection from your {SmartProxy} to the upstream repo is faster than from your {ProjectServer}.

Reviewer: The current wording makes it sound like simplified ACSs are the only way to speed up smart proxy syncs. Maybe:

```suggestion
Simplified alternate content sources are ideal for situations where the connection from your {SmartProxy} to the upstream repo is faster than from your {ProjectServer}.
```

ifdef::satellite[]
Selecting the Red Hat products when creating a simplified alternate content source will download the content to the {SmartProxies} from the {Team} CDN.
endif::[]

RHUI::

> RHUI alternate content source downloads content from the Red Hat Update Infrastructure server in your environment.

Reviewer: To keep things consistent:

```suggestion
RHUI alternate content sources download content from a Red Hat Update Infrastructure server. Help with finding the correct network paths and importing authentication credentials is provided in the user interface.
```

Author: > Help with finding the correct network paths and importing authentication credentials is provided in the user interface.

@ianballou let me check with @maximiliankolb about adding this because as I see it, this part is very evident in the UI, so do we need to mention it in the docs explicitly? What do you think @maximiliankolb

Author: @maximiliankolb yep, here's [a link to the demo, the pertinent section starts at around 13:38](https://www.youtube.com/watch?v=k7naZLzgFro). Let me know what you think.

Reviewer 3: Maybe change to

````
RHUI alternate content sources download content from a Red Hat Update Infrastructure server.
{ProjectWebUI} provides examples to help you find the network paths and to import authentication credentials.
````

ifdef::satellite[]

> {Team} only supports RHUI type alternate content source with default RHUI installations of version RHUI 4 or above.

Reviewer: Maybe something like:

```suggestion
RHUI alternate content sources used by {Team} must be version 4 or greater and use the default installation configuration. For example, AWS RHUI is unsupported because it uses an installation scenario with unique authentication requirements.
```

Author: @ianballou {Team} resolves to `Red Hat` for Satellite; `Foreman` for Foreman, etc, so the first sentence renders as below for Satellite:
`Red Hat only supports RHUI type alternate content source with default RHUI installations of version RHUI 4 or above.` 

Does this look better:
`The RHUI alternate content source must be RHUI version 4 or greater and use the default installation configuration. For example, AWS RHUI is unsupported because it uses an installation scenario with unique authentication requirements.`

Reviewer: Looks great!

> The RHUI alternate content source must be RHUI version 4 or greater and use the default installation configuration. For example, AWS RHUI is unsupported because it uses an installation scenario with unique authentication requirements.

Reviewer 3: ```suggestion
The RHUI alternate content source must be RHUI version 4 or greater and use the default installation configuration.
For example, AWS RHUI is unsupported because it uses an installation scenario with unique authentication requirements.
```

ifdef::satellite[]
Selecting the Red Hat products when creating a simplified alternate content source will download the content to the {SmartProxies} from the {Team} CDN.
endif::[]

RHUI::
RHUI alternate content sources download content from a Red Hat Update Infrastructure server.
{ProjectWebUI} provides examples to help you find the network paths and to import authentication credentials.
ifdef::satellite[]
The RHUI alternate content source must be RHUI version 4 or greater and use the default installation configuration.
For example, AWS RHUI is unsupported because it uses an installation scenario with unique authentication requirements.
endif::[]

.Permission Requirements for Alternate Content Sources

Non-administrator users must have the below permissions to manage alternate content sources:

. `view_smart_proxies`
. `view_content_credentials`
. `view_organizations`
. `view_products`

In addition to the above permissions, assign permissions specific to alternate content sources, depending on the actions the users can perform:

. `view_alternate_content_sources`
. `create_alternate_content_sources`
. `edit_alternate_content_sources`

> . `destroy_alternate_content_sources`

Reviewer 3: It would be nice if this list was sorted alphabetically.
